llvm/include/llvm/CodeGen/MachineScheduler.h

```diff
@@ -918,7 +918,7 @@ class SchedBoundary {
   };
 
   ScheduleDAGMI *DAG = nullptr;
-  const MachineSchedStrategy *SchedImpl = nullptr;
+  MachineSchedStrategy *SchedImpl = nullptr;
```

Comment:
It makes sense, but it looks a bit fragile.

Reply:
I still need to convince myself that it is fine to have a non-const `isAvailableNode()`, but in the end I think it makes sense to allow the SchedStrategies to keep some state so as to render better decisions given the global context of the region.

Reply:
I agree it makes sense. If it weren't for legacy in the code base you might want to separate query and state update. But whatever you do, there's probably some consts to be dropped, so I think its fine.